keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/NpmArtifactSupplier.kt

```diff
@@ -26,18 +26,19 @@ class NpmArtifactSupplier(
   override val eventPublisher: EventPublisher,
   private val artifactService: ArtifactService,
   override val artifactMetadataService: ArtifactMetadataService
-) : BaseArtifactSupplier<NpmArtifact, NpmVersioningStrategy>(artifactMetadataService) {
+) : BaseArtifactSupplier<NpmArtifact, NpmVersionSortingStrategy>(artifactMetadataService) {
 
   override val supportedArtifact = SupportedArtifact(NPM, NpmArtifact::class.java)
 
-  override val supportedVersioningStrategy =
-    SupportedVersioningStrategy(NPM, NpmVersioningStrategy::class.java)
+  override val supportedSortingStrategy =
+    SupportedSortingStrategy(NPM, NpmVersionSortingStrategy::class.java)
 
   override fun getLatestArtifact(deliveryConfig: DeliveryConfig, artifact: DeliveryArtifact): PublishedArtifact? =
     runWithIoContext {
+      // FIXME: in order to sort by properties other than version below, the service needs to return additional data
```

Comment:
Another example of where we're calling an external system to retrieve artifact "versions", and will need more data in order to use the `SortingStrategy`. We could return more data in the list (bigger responses), or make N calls to fill in the details for each version (more calls).